### PR TITLE
Consumer (and producer) API endpoints

### PR DESCRIPTION
**Pull request by MikeEdgar**
Replaces #170 (new fork)

### CODE REVIEW

**Comment by MikeEdgar** [comment]
@wtrocki - do you want to take a look at the API changes and let me know if you see anything that can be improved?

**Comment by wtrocki** [comment]
@MikeEdgar since repo is public maybe we can setup owners for the `./openapi` space so we can automatically be pinged. I'm happy to do it if you think it is good idea

**Comment by MikeEdgar** [comment]
Yes, great idea!

**Comment by wtrocki** [comment]
API looks good!

**Review by rareddy — APPROVED** [review]
lgtm

**Review by grdryn — APPROVED** [review]

File: kafka-admin/src/main/java/org/bf2/admin/kafka/admin/model/Types.java [hunk 1]
```diff
@@ -1677,4 +1687,248 @@ public void setClassName(String className) {
...
+        @Schema(description = "The record's partition within the topic")
+        Integer partition;
+
+        @Schema(readOnly = true, description = "The record's offset within the topic partition")
+        Long offset;
+
+        @Schema(description = "Timestamp associated with the record. The type is indicated by `timestampType`. When producing a record, this value will be used as the record's `CREATE_TIME`.", format = "date-time")
+        String timestamp;
+
+        @Schema(readOnly = true, description = "The record's partition within the topic")
```
[the remark this answers is not in the document]
Comment (MikeEdgar) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/model/Types.java]: Thank you, fixed it

**Pull request merged**